peering/models/models.py

```diff
@@ -986,6 +1032,27 @@ def is_abandoned(self):
             return False
         return True
 
+    def generate_service_ref(self) -> str:
+        """
+        Generate a Unique Service Reference for an IX Session
+        from Local ASN with 6 digit hex UUID. Example: IX9268-FD130FS/I<asn>-<hex>S
+
+        Return:
+            str: Service Reference
+        """
+        asn = self.ixp_connection.internet_exchange_point.local_autonomous_system.asn
+        return "IX{0}-{1}S".format(asn, uuid.uuid4().hex[:6].upper())
+
+    def save(self, *args, **kwargs):
+        """
+        Overwrite Model Save to generate a Unique Service Reference
+        if left blank on Save/Update
+        """
+        if self.service_reference is None:
+            self.service_reference = self.generate_service_ref()
+
+        return super().save(*args, **kwargs)
+
```

Comment:
Since this code is more or less the same for both session types, consider putting it into the abstract `BGPSession` class, even if the method to generate a reference must be implemented in subclasses.